models/productDao.js

```diff
@@ -3,6 +3,29 @@ const appDataSource = require('./appDataSource');
 const conditionMake = require('./conditionMake');
 const { DatabaseError } = require('../middlewares/error');
 
+const searchProduct = async () => {
+  try {
+    return await appDataSource.query(
+      `SELECT
+        p.id,
+        p.names AS name,
+        p.sub_description,
+        sub_category.title AS subCategory,
+        main_category.title AS mainCategory,
+        image.image_url
+        FROM product p
+        JOIN sub_category
+        ON p.sub_category_id = sub_category.id
+        JOIN main_category
+        ON sub_category.main_category_id = main_category.id
+        JOIN  (SELECT product_id, JSON_ARRAYAGG(image_url) AS image_url FROM product_image GROUP BY product_id) AS image
```

Comment:
- 가독성을 위해 전체적으로 라인브레이킹을 한 번 고민해보세요! 
```suggestion
        JOIN  (
            SELECT product_id, 
            JSON_ARRAYAGG(image_url) AS image_url 
            FROM product_image 
            GROUP BY product_id 
        )  AS image
```